### PR TITLE
hagen poiseuille equation firtst commit

### PR DESCRIPTION
**Pull request by aokiyoshi**

### CODE REVIEW

File: symplyphysics/laws/hydro/hagen_poiseuille_equation.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+@validate_input(
+    dynamic_viscosity_=dynamic_viscosity,
+    length_=length,
+    flow_rate_=flow_rate,
+    radius_=radius,
+)
+@validate_output(delta_pressure)
+def calculate_delta_p(
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hagen_poiseuille_equation.py:38]: Don't shortcut. Use full name.

File: symplyphysics/laws/hydro/hagen_poiseuille_equation.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+from symplyphysics import (Quantity, Symbol, print_expression, units,
+                           validate_input, validate_output)
+
+# Law: delta_p = 8 * mu * L * Q / (pi * R**4)
+# delta_p - pressure difference,
+# mu is viscosity,
+# L is length,
+# Q is the volumetric flow rate,
+# R is the pipe radius
+# A is cross-section area of pipe.
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hagen_poiseuille_equation.py:14]: I do not see it being used.

File: symplyphysics/laws/hydro/hagen_poiseuille_equation.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from math import pi
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hagen_poiseuille_equation.py:1]: Use sympy pi.

File: test/hydro/hagen_poiseuille_equation_test.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    q = Quantity(1 * (units.meter**3 / units.second))
+    r = Quantity(0.1 * units.meter)
+    Args = namedtuple("Args", ["mu", "l", "q", "r"])
+    return Args(mu=mu, l=l, q=q, r=r)
+
+
+def test_basic(test_args):
+    result = hagen_poiseuille_equation.calculate_delta_p(test_args.mu, test_args.l, test_args.q, test_args.r)
+    result_pressure = convert_to(result, units.pascal)
+    assert result_pressure == approx(2.276, 0.001)
```
Comment (blackyblack) [inline test/hydro/hagen_poiseuille_equation_test.py:20]: It is advised to use some data from the external sources to verify the law. Only use synthetic data if there are no other examples, or if calculations are trivial.

**Review by blackyblack — APPROVED** [review]

File: test/hydro/hagen_poiseuille_equation_test.py (hunk 4 above)
Comment (aokiyoshi) [reply test/hydro/hagen_poiseuille_equation_test.py:20]: Isn't it trival calculation?

Comment (blackyblack) [reply test/hydro/hagen_poiseuille_equation_test.py:20]: Might be easier. This time I approve it, but next time check for external sources first.

File: symplyphysics/laws/hydro/hagen_poiseuille_equation.py [hunk 5]
```diff
@@ -21,7 +18,7 @@
 
 
 law = Eq(delta_pressure, 8 * dynamic_viscosity *
-         length * flow_rate / (pi * radius**4))
+         length * flow_rate / (pi.round(4) * radius**4))
```
Comment (blackyblack) [inline symplyphysics/laws/hydro/hagen_poiseuille_equation.py:21]: Don't round pi.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**